src/loops.rs

```diff
@@ -264,22 +250,20 @@ impl LateLintPass for LoopsPass {
                                 if in_external_macro(cx, expr.span) {
                                     return;
                                 }
-                                let loop_body = if inner_stmt_expr.is_some() {
-                                    // FIXME: should probably be an ellipsis
-                                    // tabbing and newline is probably a bad idea, especially for large blocks
-                                    Cow::Owned(format!("{{\n    {}\n}}", other_stuff.join("\n    ")))
-                                } else {
-                                    expr_block(cx, &arms[0].body, Some(other_stuff.join("\n    ")), "..")
-                                };
+
+                                // NOTE: we used to make build a body here instead of using
+                                // ellipsis, this was remove because:
+                                // 1) it was ugly with big bodies;
+                                // 2) it was not indented properly;
+                                // 3) it wasn’t very smart (see #675).
                                 span_lint_and_then(cx,
                                                    WHILE_LET_LOOP,
                                                    expr.span,
                                                    "this loop could be written as a `while let` loop",
                                                    |db| {
-                                                       let sug = format!("while let {} = {} {}",
+                                                       let sug = format!("while let {} = {} {{ ... }}",
```

Comment:
Microscopical nit: we customarily use `..` to denote multiple statements (or fields).